storage/innobase/xtrabackup/doc/source/security/pxb-selinux.rst

```diff
@@ -0,0 +1,119 @@
+
+|Percona XtraBackup| works locally. The service is not network-facing and cannot be exploited externally. The service interacts only with the local user, who provides the parameters. |Percona XtraBackup| requires access to the ``target-dir`` location. 
+
+Confine XtraBackup
+--------------------
+
+You can modify your security configuration to confine |Percona XtraBackup|. The first question is where to store the backup files. The service requires read and write access to the selected location. 
+
+You can use either of the following methods:
+
+* Allow |Percona XtraBackup| to write to any location. The user provides any path to the ``target-dir`` parameter. 
+
+* Allow |Percona XtraBackup| to write to a specific location, such as /backups or the user's home directory. 
+
+The first option opens the entire system to read and write. Select the second option to harden your security.
+
+Install SELinux tools 
+----------------------
+
+To work with policies, you must install the SELinux tools. To find which package provides the ``semanage`` command and install the package. The following is an example on CentOS 7. 
+
+    .. sourcecode:: bash
+
+        $ yum provides *bin/semanage
+        ...
+        policycoreutils-python-2.5-34.el7.x86_64 : SELinux policy core python utilities
+        ...
+        $ sudo yum install -y policycoreutils-python
+
+The following is an example on CentOS 8:
+
+    .. sourcecode:: bash
+
+        $ yum provides *bin/semanage
+        ...
+        policycoreutils-python-utils-2.8-16.1.el8.noarch : SELinux policy core python utilities
+        ...
+        $ sudo yum install -y policycoreutils-python-utils
+
+Create a policy
+-----------------
+
+Use a modular approach to create an SELinux policy. Create a policy module to manage XtraBackup. You must create a ``.te`` file for type enforcement, and an optional ``.fc`` file for the file contexts. 
+
+
+Use `ps -efZ | grep xtrabackup` to verify the service is not confined by SELinux.
+
+Create the ``xtrabackup.fc`` file and add content. This file defines the security contexts. 
+
+    .. sourcecode:: text
+
+        /usr/bin/xtrabackup    -- gen_context(system_u:object_r:xtrabackup_exec_t,s0)
+        /usr/bin/xbcrypt    -- gen_context(system_u:object_r:xtrabackup_exec_t,s0)
+        /usr/bin/xbstream    -- gen_context(system_u:object_r:xtrabackup_exec_t,s0)
+        /usr/bin/xbcloud    -- gen_context(system_u:object_r:xtrabackup_exec_t,s0)
+        /backups(/.*)?       system_u:object_r:xtrabackup_data_t:s0
+
+.. note:: If you are using the ``/backups`` directory you must have the last line. If you are storing the backups in the user's home directory, you can omit this line.
+
+Download the ``xtrabackup.tc`` file from the following location:
```

Comment:
It should be xtrabackup.te instead of xtrabackup.tc

Reply:
fixed.